### ISSUE TITLE
Anonsurf got problems to install

### ISSUE THREAD

**Issue by R3TR0NIX** [Und3rf10w/kali-anonsurf#19]
So I am trying to download kali-anonsurf, cloned the git project with git clone, did ./installer.sh and it got this:
```
./installer.sh: line 13: /etc/apt/sources.list.d/i2p.list: No such file or directory
--2016-11-30 20:16:38--  https://geti2p.net/_static/i2p-debian-repo.key.asc
Resolving geti2p.net (geti2p.net)... 91.143.92.136, 2a02:180:a:65:2456:6542:1101:1010
Connecting to geti2p.net (geti2p.net)|91.143.92.136|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 14455 (14K) [text/plain]
Saving to: ‘/tmp/i2p-debian-repo.key.asc’

/tmp/i2p-debian-rep 100%[===================>]  14.12K  --.-KB/s    in 0.1s    

2016-11-30 20:16:39 (142 KB/s) - ‘/tmp/i2p-debian-repo.key.asc’ saved [14455/14455]

OK
Get:1 http://kali.mirror.garr.it/mirrors/kali kali-rolling InRelease [30.5 kB]
Fetched 30.5 kB in 1s (19.1 kB/s)                     
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libservlet3.0-java is already the newest version (7.0.73-1).
0 upgraded, 0 newly installed, 0 to remove and 450 not upgraded.
--2016-11-30 20:17:08--  http://ftp.us.debian.org/debian/pool/main/j/jetty8/libjetty8-java_8.1.16-4_all.deb
Resolving ftp.us.debian.org (ftp.us.debian.org)... 208.80.154.15, 128.61.240.89, 64.50.233.100, ...
Connecting to ftp.us.debian.org (ftp.us.debian.org)|208.80.154.15|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1666404 (1.6M) [application/octet-stream]
Saving to: ‘libjetty8-java_8.1.16-4_all.deb’

libjetty8-java_8.1. 100%[===================>]   1.59M   419KB/s    in 4.2s    

2016-11-30 20:17:13 (390 KB/s) - ‘libjetty8-java_8.1.16-4_all.deb’ saved [1666404/1666404]

(Reading database ... 349164 files and directories currently installed.)
Preparing to unpack libjetty8-java_8.1.16-4_all.deb ...
Unpacking libjetty8-java (8.1.16-4) over (8.1.16-4) ...
Setting up libjetty8-java (8.1.16-4) ...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package i2p
E: Unable to locate package i2p-router
E: Unable to locate package libjbigi-jni
Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 450 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package i2p-keyring
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package i2p
dpkg-deb: building package 'kali-anonsurf' in 'kali-anonsurf.deb'.
Selecting previously unselected package kali-anonsurf.
(Reading database ... 349164 files and directories currently installed.)
Preparing to unpack kali-anonsurf.deb ...
Unpacking kali-anonsurf (1.2.2.1) ...
dpkg: dependency problems prevent configuration of kali-anonsurf:
 kali-anonsurf depends on secure-delete; however:
  Package secure-delete is not installed.
 kali-anonsurf depends on tor; however:
  Package tor is not installed.
 kali-anonsurf depends on i2p; however:
  Package i2p is not installed.

dpkg: error processing package kali-anonsurf (--install):
 dependency problems - leaving unconfigured
Processing triggers for systemd (232-3) ...
Errors were encountered while processing:
 kali-anonsurf
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following package was automatically installed and is no longer required:
  secure-delete
Use 'apt autoremove' to remove it.
The following additional packages will be installed:
  secure-delete tor tor-geoipdb torsocks
Suggested packages:
  mixmaster torbrowser-launcher tor-arm apparmor-utils obfsproxy obfs4proxy
The following packages will be REMOVED:
  kali-anonsurf
The following NEW packages will be installed:
  secure-delete tor tor-geoipdb torsocks
0 upgraded, 4 newly installed, 1 to remove and 450 not upgraded.
1 not fully installed or removed.
Need to get 2,569 kB of archives.
After this operation, 10.0 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 secure-delete amd64 3.1-6 [69.6 kB]
Get:2 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 tor amd64 0.2.8.9-1 [1,399 kB]
Get:3 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 torsocks amd64 2.2.0-1 [72.5 kB]
Get:4 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 tor-geoipdb all 0.2.8.9-1 [1,028 kB]
Fetched 2,569 kB in 8s (296 kB/s)                                              
(Reading database ... 349172 files and directories currently installed.)
Removing kali-anonsurf (1.2.2.1) ...
Selecting previously unselected package secure-delete.
(Reading database ... 349164 files and directories currently installed.)
Preparing to unpack .../0-secure-delete_3.1-6_amd64.deb ...
Unpacking secure-delete (3.1-6) ...
Selecting previously unselected package tor.
Preparing to unpack .../1-tor_0.2.8.9-1_amd64.deb ...
Unpacking tor (0.2.8.9-1) ...
Selecting previously unselected package torsocks.
Preparing to unpack .../2-torsocks_2.2.0-1_amd64.deb ...
Unpacking torsocks (2.2.0-1) ...
Selecting previously unselected package tor-geoipdb.
Preparing to unpack .../3-tor-geoipdb_0.2.8.9-1_all.deb ...
Unpacking tor-geoipdb (0.2.8.9-1) ...
Setting up tor (0.2.8.9-1) ...
useradd: existing lock file /etc/passwd.lock without a PID
useradd: cannot lock /etc/passwd; try again later.
adduser: `/usr/sbin/useradd -d /var/lib/tor -g debian-tor -s /bin/false -u 138 debian-tor' returned error code 1. Exiting.
dpkg: error processing package tor (--configure):
 subprocess installed post-installation script returned error exit status 1
Setting up secure-delete (3.1-6) ...
dpkg: dependency problems prevent configuration of tor-geoipdb:
 tor-geoipdb depends on tor (>= 0.2.8.9-1); however:
  Package tor is not configured yet.

dpkg: error processing package tor-geoipdb (--configure):
 dependency problems - leaving unconfigured
Processing triggers for systemd (232-3) ...
Processing triggers for man-db (2.7.5-1) ...
Setting up torsocks (2.2.0-1) ...
Errors were encountered while processing:
 tor
 tor-geoipdb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

I saw Issue #15 and tried to do it, but it doesn't find the i2p-* packages... Any help?

**Comment by Und3rf10w** [comment]
For some reason it wasn't able to write the file ```/etc/apt/sources.list.d/i2p.list```

To resolve this, as root, please run: ```touch /etc/apt/sources.list.d/i2p.list```, then rerun the installer.

**Comment by R3TR0NIX** [comment]
There wasn't any folder so i created my own sources.list.d/i2p.list and leaved it blank. then i ran the command touch /etc/apt/sources.list.d/i2p.list, removed the kali-anonsurf.deb and ran the installer. now it showed this:
--2016-12-01 16:45:58--  https://geti2p.net/_static/i2p-debian-repo.key.asc
Resolving geti2p.net (geti2p.net)... 91.143.92.136, 2a02:180:a:65:2456:6542:1101:1010
Connecting to geti2p.net (geti2p.net)|91.143.92.136|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 14455 (14K) [text/plain]
Saving to: ‘/tmp/i2p-debian-repo.key.asc’

/tmp/i2p-debian-repo.k 100%[===========================>]  14.12K  --.-KB/s    in 0.08s   

2016-12-01 16:45:59 (185 KB/s) - ‘/tmp/i2p-debian-repo.key.asc’ saved [14455/14455]

OK
Get:2 http://deb.i2p2.no unstable InRelease [12.1 kB]                        
Get:3 http://deb.i2p2.no unstable/main amd64 Packages [3,224 B]
Get:4 http://deb.i2p2.no unstable/main i386 Packages [3,092 B]                     
Get:1 http://kali.mirror.garr.it/mirrors/kali kali-rolling InRelease [30.5 kB]         
Get:5 http://kali.mirror.garr.it/mirrors/kali kali-rolling/contrib Sources [66.6 kB]
Get:6 http://kali.mirror.garr.it/mirrors/kali kali-rolling/non-free Sources [126 kB]
Get:7 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main Sources [10.7 MB]
Get:8 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main i386 Packages [14.4 MB]   
Get:9 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 Packages [14.4 MB]  
Get:10 http://kali.mirror.garr.it/mirrors/kali kali-rolling/contrib i386 Packages [101 kB]
Get:11 http://kali.mirror.garr.it/mirrors/kali kali-rolling/contrib amd64 Packages [104 kB]
Get:12 http://kali.mirror.garr.it/mirrors/kali kali-rolling/non-free amd64 Packages [163 kB]
Get:13 http://kali.mirror.garr.it/mirrors/kali kali-rolling/non-free i386 Packages [144 kB]
Fetched 40.3 MB in 26s (1,532 kB/s)                                                       
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libservlet3.0-java is already the newest version (7.0.73-1).
The following package was automatically installed and is no longer required:
  secure-delete
Use 'apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 473 not upgraded.
2 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n] y
Setting up tor (0.2.8.9-1) ...
useradd: existing lock file /etc/passwd.lock without a PID
useradd: cannot lock /etc/passwd; try again later.
adduser: `/usr/sbin/useradd -d /var/lib/tor -g debian-tor -s /bin/false -u 138 debian-tor' returned error code 1. Exiting.
dpkg: error processing package tor (--configure):
 subprocess installed post-installation script returned error exit status 1
dpkg: dependency problems prevent configuration of tor-geoipdb:
 tor-geoipdb depends on tor (>= 0.2.8.9-1); however:
  Package tor is not configured yet.

dpkg: error processing package tor-geoipdb (--configure):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 tor
 tor-geoipdb
E: Sub-process /usr/bin/dpkg returned an error code (1)
--2016-12-01 16:49:22--  http://ftp.us.debian.org/debian/pool/main/j/jetty8/libjetty8-java_8.1.16-4_all.deb
Resolving ftp.us.debian.org (ftp.us.debian.org)... 128.30.2.26, 128.61.240.89, 208.80.154.15, ...
Connecting to ftp.us.debian.org (ftp.us.debian.org)|128.30.2.26|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1666404 (1.6M) [application/x-debian-package]
Saving to: ‘libjetty8-java_8.1.16-4_all.deb.1’

libjetty8-java_8.1.16- 100%[===========================>]   1.59M   711KB/s    in 2.3s    

2016-12-01 16:49:25 (711 KB/s) - ‘libjetty8-java_8.1.16-4_all.deb.1’ saved [1666404/1666404]

(Reading database ... 349244 files and directories currently installed.)
Preparing to unpack libjetty8-java_8.1.16-4_all.deb ...
Unpacking libjetty8-java (8.1.16-4) over (8.1.16-4) ...
Setting up libjetty8-java (8.1.16-4) ...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libservlet3.0-java is already the newest version (7.0.73-1).
The following package was automatically installed and is no longer required:
  secure-delete
Use 'apt autoremove' to remove it.
The following additional packages will be installed:
  libservice-wrapper-java libservice-wrapper-jni service-wrapper ttf-dejavu-core
  ttf-dejavu-extra
Suggested packages:
  apparmor i2p-messenger i2phex i2p-tahoe-lafs imule irc-client itoopie mail-client
  news-reader polipo privoxy robert syndie xul-ext-torbutton ecj ant libecj-java-gcj
  libgetopt-java-doc libservice-wrapper-doc
The following NEW packages will be installed:
  glassfish-javaee i2p i2p-router libecj-java libgetopt-java libjbigi-jni
  libservice-wrapper-java libservice-wrapper-jni service-wrapper ttf-dejavu
  ttf-dejavu-core ttf-dejavu-extra
0 upgraded, 12 newly installed, 0 to remove and 473 not upgraded.
2 not fully installed or removed.
Need to get 11.9 MB of archives.
After this operation, 15.8 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://deb.i2p2.no unstable/main amd64 i2p-router all 0.9.27-1~deb8u+1 [8,597 kB]
Get:2 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 glassfish-javaee all 1:2.1.1-b31g+dfsg1-3 [839 kB]
Get:3 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 libecj-java all 3.11.0-7 [1,695 kB]
Get:4 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 libgetopt-java all 1.0.14+dfsg-3 [24.9 kB]
Get:5 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 libservice-wrapper-java all 3.5.26-3 [320 kB]
Get:6 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 libservice-wrapper-jni amd64 3.5.26-3 [86.7 kB]
Get:7 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 service-wrapper amd64 3.5.26-3 [176 kB]
Get:8 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 ttf-dejavu-core all 2.37-1 [32.1 kB]
Get:9 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 ttf-dejavu-extra all 2.37-1 [32.2 kB]
Get:10 http://kali.mirror.garr.it/mirrors/kali kali-rolling/main amd64 ttf-dejavu all 2.37-1 [32.0 kB]
Get:11 http://deb.i2p2.no unstable/main amd64 libjbigi-jni amd64 0.9.27-1~deb8u+1 [4,962 B]
Get:12 http://deb.i2p2.no unstable/main amd64 i2p all 0.9.27-1~deb8u+1 [47.1 kB]          
Fetched 11.9 MB in 13s (864 kB/s)                                                         
Preconfiguring packages ...
Selecting previously unselected package glassfish-javaee.
(Reading database ... 349244 files and directories currently installed.)
Preparing to unpack .../00-glassfish-javaee_1%3a2.1.1-b31g+dfsg1-3_all.deb ...
Unpacking glassfish-javaee (1:2.1.1-b31g+dfsg1-3) ...
Selecting previously unselected package libecj-java.
Preparing to unpack .../01-libecj-java_3.11.0-7_all.deb ...
Unpacking libecj-java (3.11.0-7) ...
Selecting previously unselected package libgetopt-java.
Preparing to unpack .../02-libgetopt-java_1.0.14+dfsg-3_all.deb ...
Unpacking libgetopt-java (1.0.14+dfsg-3) ...
Selecting previously unselected package i2p-router.
Preparing to unpack .../03-i2p-router_0.9.27-1~deb8u+1_all.deb ...
Unpacking i2p-router (0.9.27-1~deb8u+1) ...
Selecting previously unselected package libjbigi-jni.
Preparing to unpack .../04-libjbigi-jni_0.9.27-1~deb8u+1_amd64.deb ...
Unpacking libjbigi-jni (0.9.27-1~deb8u+1) ...
Selecting previously unselected package libservice-wrapper-java.
Preparing to unpack .../05-libservice-wrapper-java_3.5.26-3_all.deb ...
Unpacking libservice-wrapper-java (3.5.26-3) ...
Selecting previously unselected package libservice-wrapper-jni.
Preparing to unpack .../06-libservice-wrapper-jni_3.5.26-3_amd64.deb ...
Unpacking libservice-wrapper-jni (3.5.26-3) ...
Selecting previously unselected package service-wrapper.
Preparing to unpack .../07-service-wrapper_3.5.26-3_amd64.deb ...
Unpacking service-wrapper (3.5.26-3) ...
Selecting previously unselected package i2p.
Preparing to unpack .../08-i2p_0.9.27-1~deb8u+1_all.deb ...
Unpacking i2p (0.9.27-1~deb8u+1) ...
Selecting previously unselected package ttf-dejavu-core.
Preparing to unpack .../09-ttf-dejavu-core_2.37-1_all.deb ...
Unpacking ttf-dejavu-core (2.37-1) ...
Selecting previously unselected package ttf-dejavu-extra.
Preparing to unpack .../10-ttf-dejavu-extra_2.37-1_all.deb ...
Unpacking ttf-dejavu-extra (2.37-1) ...
Selecting previously unselected package ttf-dejavu.
Preparing to unpack .../11-ttf-dejavu_2.37-1_all.deb ...
Unpacking ttf-dejavu (2.37-1) ...
Setting up ttf-dejavu-core (2.37-1) ...
Setting up tor (0.2.8.9-1) ...
useradd: existing lock file /etc/passwd.lock without a PID
useradd: cannot lock /etc/passwd; try again later.
adduser: `/usr/sbin/useradd -d /var/lib/tor -g debian-tor -s /bin/false -u 138 debian-tor' returned error code 1. Exiting.
dpkg: error processing package tor (--configure):
 subprocess installed post-installation script returned error exit status 1
Setting up libservice-wrapper-jni (3.5.26-3) ...
Setting up libservice-wrapper-java (3.5.26-3) ...
dpkg: dependency problems prevent configuration of tor-geoipdb:
 tor-geoipdb depends on tor (>= 0.2.8.9-1); however:
  Package tor is not configured yet.

dpkg: error processing package tor-geoipdb (--configure):
 dependency problems - leaving unconfigured
Setting up libecj-java (3.11.0-7) ...
Setting up ttf-dejavu-extra (2.37-1) ...
Processing triggers for systemd (232-3) ...
Processing triggers for man-db (2.7.5-1) ...
Setting up glassfish-javaee (1:2.1.1-b31g+dfsg1-3) ...
Setting up service-wrapper (3.5.26-3) ...
Setting up libgetopt-java (1.0.14+dfsg-3) ...
Processing triggers for fontconfig (2.11.0-6.7) ...
Setting up ttf-dejavu (2.37-1) ...
Setting up i2p-router (0.9.27-1~deb8u+1) ...
Setting up libjbigi-jni (0.9.27-1~deb8u+1) ...
Setting up i2p (0.9.27-1~deb8u+1) ...
Synchronizing state of i2p.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install disable i2p
update-rc.d: We have no instructions for the i2p init script.
update-rc.d: It looks like a network service, we disable it.
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
dpkg: error processing package i2p (--configure):
 subprocess installed post-installation script returned error exit status 1
Processing triggers for systemd (232-3) ...
Errors were encountered while processing:
 tor
 tor-geoipdb
 i2p
E: Sub-process /usr/bin/dpkg returned an error code (1)
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following package was automatically installed and is no longer required:
  secure-delete
Use 'apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 473 not upgraded.
3 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Setting up tor (0.2.8.9-1) ...
useradd: existing lock file /etc/passwd.lock without a PID
useradd: cannot lock /etc/passwd; try again later.
adduser: `/usr/sbin/useradd -d /var/lib/tor -g debian-tor -s /bin/false -u 138 debian-tor' returned error code 1. Exiting.
dpkg: error processing package tor (--configure):
 subprocess installed post-installation script returned error exit status 1
dpkg: dependency problems prevent configuration of tor-geoipdb:
 tor-geoipdb depends on tor (>= 0.2.8.9-1); however:
  Package tor is not configured yet.

dpkg: error processing package tor-geoipdb (--configure):
 dependency problems - leaving unconfigured
Setting up i2p (0.9.27-1~deb8u+1) ...
Synchronizing state of i2p.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install disable i2p
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
dpkg: error processing package i2p (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 tor
 tor-geoipdb
 i2p
E: Sub-process /usr/bin/dpkg returned an error code (1)
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following package was automatically installed and is no longer required:
  secure-delete
Use 'apt autoremove' to remove it.
The following additional packages will be installed:
  killyourtv-keyring
The following NEW packages will be installed:
  i2p-keyring killyourtv-keyring
0 upgraded, 2 newly installed, 0 to remove and 473 not upgraded.
3 not fully installed or removed.
Need to get 25.3 kB of archives.
After this operation, 71.7 kB of additional disk space will be used.
Get:1 http://deb.i2p2.no unstable/main amd64 i2p-keyring all 2016.10.01 [12.4 kB]
Get:2 http://deb.i2p2.no unstable/main amd64 killyourtv-keyring all 2015.03.27.1 [12.9 kB]
Fetched 25.3 kB in 0s (29.3 kB/s)            
Selecting previously unselected package i2p-keyring.
(Reading database ... 350271 files and directories currently installed.)
Preparing to unpack .../0-i2p-keyring_2016.10.01_all.deb ...
Unpacking i2p-keyring (2016.10.01) ...
Selecting previously unselected package killyourtv-keyring.
Preparing to unpack .../1-killyourtv-keyring_2015.03.27.1_all.deb ...
Unpacking killyourtv-keyring (2015.03.27.1) ...
Setting up tor (0.2.8.9-1) ...
useradd: existing lock file /etc/passwd.lock without a PID
useradd: cannot lock /etc/passwd; try again later.
adduser: `/usr/sbin/useradd -d /var/lib/tor -g debian-tor -s /bin/false -u 138 debian-tor' returned error code 1. Exiting.
dpkg: error processing package tor (--configure):
 subprocess installed post-installation script returned error exit status 1
Setting up i2p-keyring (2016.10.01) ...
Adding/refreshing keys:
  67ECE5605BCF1346 1381446626  [ expires 1506878164 ]
... Warning: apt-key should not be used in scripts (called from postinst maintainerscript of the package i2p-keyring)
OK
dpkg: dependency problems prevent configuration of tor-geoipdb:
 tor-geoipdb depends on tor (>= 0.2.8.9-1); however:
  Package tor is not configured yet.

dpkg: error processing package tor-geoipdb (--configure):
 dependency problems - leaving unconfigured
Setting up i2p (0.9.27-1~deb8u+1) ...
Synchronizing state of i2p.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install disable i2p
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
dpkg: error processing package i2p (--configure):
 subprocess installed post-installation script returned error exit status 1
Setting up killyourtv-keyring (2015.03.27.1) ...
Adding/refreshing keys:
  8CF6E896B3C01B09 1365265899  [ expires 1460252357 ]
... Warning: apt-key should not be used in scripts (called from postinst maintainerscript of the package killyourtv-keyring)
OK
Removing old keys (if exist):
Errors were encountered while processing:
 tor
 tor-geoipdb
 i2pE: Sub-process /usr/bin/dpkg returned an error code (1)
Reading package lists... Done
Building dependency tree       
Reading state information... Done
secure-delete is already the newest version (3.1-6).
secure-delete set to manually installed.
tor is already the newest version (0.2.8.9-1).
tor set to manually installed.
i2p is already the newest version (0.9.27-1~deb8u+1).
0 upgraded, 0 newly installed, 0 to remove and 473 not upgraded.
3 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Setting up tor (0.2.8.9-1) ...
useradd: existing lock file /etc/passwd.lock without a PID
useradd: cannot lock /etc/passwd; try again later.
adduser: `/usr/sbin/useradd -d /var/lib/tor -g debian-tor -s /bin/false -u 138 debian-tor' returned error code 1. Exiting.
dpkg: error processing package tor (--configure):
 subprocess installed post-installation script returned error exit status 1
dpkg: dependency problems prevent configuration of tor-geoipdb:
 tor-geoipdb depends on tor (>= 0.2.8.9-1); however:
  Package tor is not configured yet.

dpkg: error processing package tor-geoipdb (--configure):
 dependency problems - leaving unconfigured
Setting up i2p (0.9.27-1~deb8u+1) ...
Synchronizing state of i2p.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install disable i2p
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
dpkg: error processing package i2p (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 tor
 tor-geoipdb
 i2p
E: Sub-process /usr/bin/dpkg returned an error code (1)
dpkg-deb: building package 'kali-anonsurf' in 'kali-anonsurf.deb'.
(Reading database ... 350281 files and directories currently installed.)
Preparing to unpack kali-anonsurf.deb ...
Unpacking kali-anonsurf (1.2.2.1) ...
dpkg: dependency problems prevent configuration of kali-anonsurf:
 kali-anonsurf depends on tor; however:
  Package tor is not configured yet.
 kali-anonsurf depends on i2p; however:
  Package i2p is not configured yet.

dpkg: error processing package kali-anonsurf (--install):
 dependency problems - leaving unconfigured
Processing triggers for systemd (232-3) ...
Errors were encountered while processing:
 kali-anonsurf
Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 473 not upgraded.
4 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Setting up tor (0.2.8.9-1) ...
useradd: existing lock file /etc/passwd.lock without a PID
useradd: cannot lock /etc/passwd; try again later.
adduser: `/usr/sbin/useradd -d /var/lib/tor -g debian-tor -s /bin/false -u 138 debian-tor' returned error code 1. Exiting.
dpkg: error processing package tor (--configure):
 subprocess installed post-installation script returned error exit status 1
dpkg: dependency problems prevent configuration of kali-anonsurf:
 kali-anonsurf depends on tor; however:
  Package tor is not configured yet.

dpkg: error processing package kali-anonsurf (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of tor-geoipdb:
 tor-geoipdb depends on tor (>= 0.2.8.9-1); however:
  Package tor is not configured yet.

dpkg: error processing package tor-geoipdb (--configure):
 dependency problems - leaving unconfigured
Setting up i2p (0.9.27-1~deb8u+1) ...
Synchronizing state of i2p.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install disable i2p
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
insserv: warning: current start runlevel(s) (empty) of script `i2p' overrides LSB defaults (2 3 4 5).
insserv: warning: current stop runlevel(s) (0 1 2 3 4 5 6) of script `i2p' overrides LSB defaults (0 1 6).
dpkg: error processing package i2p (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 tor
 kali-anonsurf
 tor-geoipdb
 i2p
E: Sub-process /usr/bin/dpkg returned an error code (1)

now what?

**Comment by Und3rf10w** [comment]
Do you have the permissions to modify ```/etc/passwd``` or ```/etc/shadow```? It looks like there's something going wrong with it. I can't imagine you wouldn't, as the installer should prevent you from using it if you don't have root permissions. It appears as though the lock files are interfering with the install of the ```tor``` package per this message:

```
Setting up tor (0.2.8.9-1) ...
useradd: existing lock file /etc/passwd.lock without a PID
useradd: cannot lock /etc/passwd; try again later.
adduser: `/usr/sbin/useradd -d /var/lib/tor -g debian-tor -s /bin/false -u 138 debian-tor' returned error code 1. Exiting.
```
See if lock files also exist for ```shadow```, ```group```, and ```gshadow```. Normally, these are in:

* ```/etc/shadow.lock```
* ```/etc/group.lock```
* ```/etc/gshadow.lock```

But you definitely need to remove ```/etc/passwd.lock```, as that's preventing you from installing ```tor``` and breaking the install.

Once you remove that, run (as root): ```apt-get install -y tor``` and that should at least install tor.

There's not enough in the logs for me to see what's going on with the ```i2p``` package though.

Can you provide the output of ```apt-get -s -V --print-uris install i2p```?

**Comment by R3TR0NIX** [comment]
EDIT: I changed the shadow.lock file to shadoww.lock (made it so it's "not real"), tried installing tor and it installed kali-anonsurf, and i2p with it. it said some things that i think were errors (but i'm not sure). I did the command "anonsurf" and it worked! then i went to ip chicked, checked my ip, started anonsurf, refreshed, and it was changed! tried to trace that ip but it said unknown! long story short, I somehow succeded it and it is working now. (yes, i tried to restart anonsurf and it still works). THANK YOU!

There is a shadow.lock, there isn't group.lock, but there is group, and group- files, and there isn't gshadow.lock, but there is gshadow and gshadow-. I removed the passwd.lock and tried installing tor, but it showed this:
E: Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)
E: Unable to lock the administration directory (/var/lib/dpkg/), is another process using it?

apt-get -s -V --print-uris install i2p:
Reading package lists... Done
Building dependency tree       
Reading state information... Done
i2p is already the newest version (0.9.27-1~deb8u+1).
0 upgraded, 0 newly installed, 0 to remove and 473 not upgraded.

**Comment by hazeek** [comment]
i have this problem from over a week can't fix please help me 

root@localhost:~# cd Desktop
root@localhost:~/Desktop# ls
kali-anonsurf
root@localhost:~/Desktop# cd kali-anonsurf
root@localhost:~/Desktop/kali-anonsurf# ls
installer.sh  kali-anonsurf-deb-src  LICENSE  README.md
root@localhost:~/Desktop/kali-anonsurf# ./installer.sh
--2017-01-20 03:02:20--  https://geti2p.net/_static/i2p-debian-repo.key.asc
Resolving geti2p.net (geti2p.net)... 91.143.92.136, 2a02:180:a:65:2456:6542:1101:1010
Connecting to geti2p.net (geti2p.net)|91.143.92.136|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 14455 (14K) [text/plain]
Saving to: ‘/tmp/i2p-debian-repo.key.asc’

/tmp/i2p-debian-rep 100%[===================>]  14.12K  35.2KB/s    in 0.4s    

2017-01-20 03:02:22 (35.2 KB/s) - ‘/tmp/i2p-debian-repo.key.asc’ saved [14455/14455]

OK
Hit:1 http://deb.i2p2.no unstable InRelease
Reading package lists... Done
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package libservlet3.0-java is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'libservlet3.0-java' has no installation candidate
--2017-01-20 03:02:25--  http://ftp.us.debian.org/debian/pool/main/j/jetty8/libjetty8-java_8.1.16-4_all.deb
Resolving ftp.us.debian.org (ftp.us.debian.org)... 208.80.154.15, 64.50.236.52, 128.30.2.26, ...
Connecting to ftp.us.debian.org (ftp.us.debian.org)|208.80.154.15|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1666404 (1.6M) [application/octet-stream]
Saving to: ‘libjetty8-java_8.1.16-4_all.deb’

libjetty8-java_8.1.16-4_ 100%[==================================>]   1.59M   296KB/s    in 7.9s    

2017-01-20 03:02:34 (207 KB/s) - ‘libjetty8-java_8.1.16-4_all.deb’ saved [1666404/1666404]

(Reading database ... 303821 files and directories currently installed.)
Preparing to unpack libjetty8-java_8.1.16-4_all.deb ...
Unpacking libjetty8-java (8.1.16-4) over (8.1.16-4) ...
dpkg: dependency problems prevent configuration of libjetty8-java:
 libjetty8-java depends on libservlet3.0-java; however:
  Package libservlet3.0-java is not installed.

dpkg: error processing package libjetty8-java (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 libjetty8-java
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package libecj-java is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package libgetopt-java is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package libservlet3.0-java is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package glassfish-javaee is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package ttf-dejavu is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  fonts-dejavu fonts-dejavu-extra fonts-dejavu-core

E: Package 'libecj-java' has no installation candidate
E: Package 'libgetopt-java' has no installation candidate
E: Package 'libservlet3.0-java' has no installation candidate
E: Package 'glassfish-javaee' has no installation candidate
E: Package 'ttf-dejavu' has no installation candidate
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following packages will be REMOVED:
  kali-anonsurf libjetty8-java
0 upgraded, 0 newly installed, 2 to remove and 0 not upgraded.
2 not fully installed or removed.
After this operation, 2,376 kB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 303821 files and directories currently installed.)
Removing kali-anonsurf (1.2.2.2) ...
Removing libjetty8-java (8.1.16-4) ...
Processing triggers for systemd (231-4) ...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
i2p-keyring is already the newest version (2016.10.01).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package tor is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package secure-delete is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'secure-delete' has no installation candidate
E: Package 'tor' has no installation candidate
dpkg-deb: error: control directory has bad permissions 700 (must be >=0755 and <=0775)
dpkg: error: cannot access archive 'kali-anonsurf.deb': No such file or directory
Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
dpkg: error: cannot access archive 'kali-anonsurf.deb': No such file or directory

**Comment by Und3rf10w** [comment]
@hazeek, please do not reopen a closed issue for this. Please open a new issue.
